packages/yarnpkg-core/sources/hashUtils.ts

```diff
@@ -25,23 +25,23 @@ export function makeHash<T extends string = string>(...args: Array<BinaryLike |
   return hash.digest(`hex`) as T;
 }
 
-export function checksumFile(path: PortablePath) {
-  return new Promise<string>((resolve, reject) => {
-    const hash = createHash(`sha512`);
-    const stream = xfs.createReadStream(path);
+export async function checksumFile(path: PortablePath) {
+  const fd = await xfs.openPromise(path, `r`);
 
-    stream.on(`data`, chunk => {
-      hash.update(chunk);
-    });
+  try {
+    const CHUNK_SIZE = 65536;
+    const chunk = Buffer.allocUnsafeSlow(CHUNK_SIZE);
```

Comment:
Why `allocUnsafeSlow` rather than `allocUnsafe`?

Reply:
Unless we change the size of the buffer pool it always allocates more than the pool size and would go to `Buffer.allocUnsafeSlow` either way

Reply:
👍